Upgrade `connectivity_plus` from v5.0.2 to v6.0.5 with updated implementations and tests

may-tas opened the pull request:
**What kind of change does this PR introduce?**

- Upgraded package `connectivity_plus` to v6.0.5

**Issue Number:**

- Fixes #2547 

**Did you add tests for your changes?**

- Yes

**Snapshots/Videos:**

- `connectivity_plus` v6.0.5 requirements :
<img width="350" height ="350" alt="requirements" src="https://github.com/user-attachments/assets/f17769c6-c732-4a70-8ad8-07c022055560" />

- tests :
<img width="364" height="550" alt="testsConnectivity" src="https://github.com/user-attachments/assets/1562e0ec-8b8f-4e56-b2be-bb6691a8abb4" />


**Summary**

- Migrated `connectivity_plus` from v5.0.2 to v6.0.5, updated implementation to meet new version requirements, and fixed related test cases.
- The major change in this migration is that `checkConnectivity()` now returns a `Future<List<ConnectivityResult>>` and `onConnectivityChanged` now returns a `Stream<List<ConnectivityResult>>` instead of a single `Future<ConnectivityResult>` and `Stream<ConnectivityResult>` respectively.

**Does this PR introduce a breaking change?**

- No

**Other information**

- I have not included the `settings.gradle`, `build.gradle`, and `gradle-wrapper.properties` files with the required changes as per the package requirements. If these are deemed necessary for this PR, I will add them in a separate commit, as they are crucial for running the package and app correctly.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

- Yes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
	- Enhanced connectivity handling to support multiple connectivity results.
	- Introduced a new method to check connectivity status with a list of results.

- **Bug Fixes**
	- Improved error handling for connectivity checks.

- **Dependencies**
	- Updated `connectivity_plus` package to version `^6.0.5`.
	- Added new dependency: `url_launcher_platform_interface` version `2.3.2`.

- **Android Configuration**
	- Updated Android project to use Java 17 and compile SDK version 34.

- **Tests**
	- Revised tests to accommodate new list-based connectivity handling.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

may-tas commented:
@palisadoes @noman2002  my android build is failing as the package requires java17 and the current gradle is not compatible.
What should be done from my end ? Please do tell me.

<img width="772" alt="issuegradle" src="https://github.com/user-attachments/assets/1d025531-aec5-4c32-bdbb-16b6e5e130c3" />

palisadoes commented:
Please fix the failing tests. Our test code coverage system will fail if any of these conditions occur:

1. The overall code coverage drops below the target threshold of the repository
2. Any file in the pull request has code coverage levels below the repository threshold
3. The code does not comply with our linting check or other related formatting requirements 

Here are some steps to take:

1. Click on the `Details` link for the tests to get an explanation of the cause(s)
2. You may have to fix a test that is not related to this PR as in some rare cases we merge faulty test code. If this is the case, fix the test in this PR. Do not open another issue.

We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

Additional notes:

1. Ensure that CodeRabbit.ai approves your work
1. Please fix any conflicting files

may-tas commented:
@palisadoes @noman2002 
- Please do tell me what should be done in this case.
- I have added all the necessary tests and it was approved by codeRabbit.
- codegov/patch is failing with a warning that 'Added line #L63 & #L110 - L112 was not covered by tests ' in connectivity_service.dart but I have already added the necessary tests for those lines.

- warning 1: 
<img width="569" alt="1Warn" src="https://github.com/user-attachments/assets/32702918-fbc4-4d9b-83e3-78c4bf82df30" />

- tests for this line :
<img width="548" alt="1Sol" src="https://github.com/user-attachments/assets/5ed6595f-bf8f-43d3-b6ac-662140cfb31e" />

- warning 2:
<img width="810" alt="2Warn" src="https://github.com/user-attachments/assets/78be87e5-4460-451b-8e1b-0e5db51f830e" />

- tests for this line :
<img width="707" alt="2Sol" src="https://github.com/user-attachments/assets/e8df0270-3885-4df3-a63b-58bf10ee0f33" />

may-tas commented:
@palisadoes I have added all the necessary tests , I am getting warning that line 63 is not covered, which I have covered if you take a look at my tests. Please tell me what to do.

I have made all the necessary changes and updated the package as it was necessary . Please review it .

may-tas commented:
@palisadoes Please review

palisadoes commented:
Please fix the failing tests

may-tas commented:
> Please fix the failing tests

Sir the java gradle is not compatible. connectivity_plus has a breaking change it will only work with java17 and above

palisadoes commented:
Please upgrade to the latest flutter, and fix the `.github/workflow/*` files so that everything works correctly

may-tas commented:
@palisadoes To pass the android build test we will have to update the Java Version in GitHub Actions Workflow.

may-tas commented:
> Please upgrade to the latest flutter/Java, and fix the `.github/workflow/*` files so that everything works correctly

Do I raise another PR for this or should I upgrade flutter to the latest version in this PR only and commit new changes.

palisadoes commented:
Please do it here. You discovered that we need to do the upgrade

may-tas commented:
@palisadoes please review, I have fixed the workflow.
Please add sensitive files label.

palisadoes commented:
Thanks. Please make a minor commit to trigger the workflow again

may-tas commented:
> Thanks. Please make a minor commit to trigger the workflow again

Doing it rn.
Thanks for your time.

may-tas commented:
@palisadoes  All done.
 please take a look.

palisadoes commented:
@may-tas You need to open another PR for this. Our push workflows are failing because the changes were not added to the `.github/workflows/push.yml` file.

palisadoes commented:
I've created an issue for it:
- https://github.com/PalisadoesFoundation/talawa/issues/2679

may-tas commented:
> I've created an issue for it:
> 
> * [Update push.yml to java-version: "17.0" #2679](https://github.com/PalisadoesFoundation/talawa/issues/2679)

Doing it rn.
